### DISCUSSION TITLE
Kernel functions vs plugins

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#381] [General]
Now that we've got plugins, I wonder if it wouldn't be better to move some of the CREXX kernel functions into them. For example substr, left, right, insert, etc.

**Comment by rvjansen** [comment]
We will know if we have done that and measured its gains. So by all means!

But you know I have misgivings about having some big system in C again; I would like to be able to still shrink the runtime part that is compiled from C source code and have as much as possible - and reasonable - in Rexx; having visions of being able to bootstrap that some day. So I would like to keep the option of having it configurable to use the plugins or the Rexx functions. Maybe by specifying that as the library.

**Comment by Peter-Jacob** [comment]
Well, it is maybe an unfair comparison.My sample uses ASCII, not Unicode. I would refrain of rebuilding it in UC.Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: René Vincent Jansen ***@***.***> Datum: 18.10.24  15:43  (GMT+01:00) An: adesutherland/CREXX ***@***.***> Cc: Peter-Jacob ***@***.***>, Author ***@***.***> Betreff: Re: [adesutherland/CREXX] Kernel functions vs plugins (Discussion #381) 
We will know if we have done that and measured its gains. So by all means!
But you know I have misgivings about having some big system in C again; I would like to be able to still shrink the runtime part that is compiled from C source code and have as much as possible - and reasonable - in Rexx; having visions of being able to bootstrap that some day. So I would like to keep the option of having it configurable to use the plugins or the Rexx functions. Maybe by specifying that as the library.
